contracts/token-manager/TokenManager.sol

```diff
@@ -182,7 +182,7 @@ abstract contract TokenManager is ITokenManager, Operatable, FlowLimit, Implemen
      * @return the amount of token actually given, which will only be different than `amount` in cases where the token takes some on-transfer fee.
      */
     function giveToken(address destinationAddress, uint256 amount) external onlyService returns (uint256) {
-        _addFlowIn(amount);
+        _addFlowIn(amount, tokenId());
```

Comment:
this is adding extra gas for every transfer. The emitter of the event will be the token manager, we can report that in the monitoring and query token id manually or in a consumer of the alert.